float32 dist
uint8 size
uint32 image_x
uint32 image_y